src/taskgraph/parameters.py

```diff
@@ -80,46 +80,76 @@ def get_version(repo_path):
 def _get_defaults(repo_root=None):
     repo_path = repo_root or os.getcwd()
     repo = get_repository(repo_path)
-    try:
-        repo_url = repo.get_url()
-        parsed_url = mozilla_repo_urls.parse(repo_url)
-        project = parsed_url.repo_name
-    except (
-        CalledProcessError,
-        mozilla_repo_urls.errors.InvalidRepoUrlError,
-        mozilla_repo_urls.errors.UnsupportedPlatformError,
-    ):
-        repo_url = ""
-        project = ""
-
-    return {
-        "base_repository": repo_url,
-        "base_ref": "",
-        "base_rev": "",
-        "build_date": int(time.time()),
-        "build_number": 1,
-        "do_not_optimize": [],
-        "enable_always_target": True,
-        "existing_tasks": {},
-        "filters": ["target_tasks_method"],
-        "head_ref": repo.branch or repo.head_rev,
-        "head_repository": repo_url,
-        "head_rev": repo.head_rev,
-        "head_tag": "",
-        "level": "3",
-        "moz_build_date": datetime.now().strftime("%Y%m%d%H%M%S"),
-        "next_version": None,
-        "optimize_strategies": None,
-        "optimize_target_tasks": True,
-        "owner": "nobody@mozilla.com",
-        "project": project,
-        "pushdate": int(time.time()),
-        "pushlog_id": "0",
-        "repository_type": repo.tool,
-        "target_tasks_method": "default",
-        "tasks_for": "",
-        "version": get_version(repo_path),
-    }
+    if repo:
```

Comment:
Thanks, this seems fine. Though rather than a big `if` statement here, let's try to only adjust the defaults for the parameters that rely on VCS. For example something like:
```python
defaults ={
    "base_repository": "SOURCE",
    "base_ref": "",
    "base_rev": "",
    "build_date": int(time.time()),
    ...
}

if repo:
    defaults.update({
        "base_repository": repo.get_url(),
        "head_ref": repo.branch or repo.head_rev,
         ...
    })
return defaults